src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -11,16 +11,11 @@
 
 namespace Roslyn.Utilities;
 
-internal static class AsyncLazy
+internal abstract class AsyncLazy<T>
```

Comment:
add note that we just have this so that people can refer to `AsyncLazy<T>` and not `AsyncLazy<T, TData>` but that we strongly discourage any other subclasses.